src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.Core.cs

```diff
@@ -55,6 +55,8 @@ internal ResourceReader(Stream stream, Dictionary<string, ResourceLocator> resCa
             "the user to only get one error.")]
         private object DeserializeObject(int typeIndex)
         {
+            Debug.Assert(Monitor.IsEntered(this));
```

Comment:
I added these to private methods that access `_store` per [this suggestion](https://github.com/dotnet/runtime/issues/74052#issuecomment-1218289577).